.github/workflows/rc_sync.yaml

```diff
@@ -61,6 +61,16 @@ jobs:
           # Exclude some files known to be divergent during the release process from the PR
           git checkout origin/main -- frontend/catalyst/_version.py
           git checkout origin/main -- doc/dev/release_notes.rst
+          git checkout origin/main -- .github/workflows/check-catalyst.yaml
+          git checkout origin/main -- .github/workflows/build-wheel-linux-x86_64.yaml
+          git checkout origin/main -- .github/workflows/build-wheel-linux-arm64.yaml
+          git checkout origin/main -- .github/workflows/build-wheel-macos-x86_64.yaml
+          git checkout origin/main -- .github/workflows/build-wheel-macos-arm64.yaml
+          git checkout origin/main -- .dep-versions
+          git checkout origin/main -- .github/workflows/scripts/linux_arm64/rh8/test_wheels.sh
+          git checkout origin/main -- Makefile
+          git checkout origin/main -- doc/requirements.txt
+          git checkout origin/main -- setup.py
```

Comment:
> @dime10 , My original motivation was to only exclude the check-catalyst.yaml file, but not too sure about the rest. I am not sure how frequently we would have changes in the other files that want in the main too. Do you think it make sense to exclude them?

Thanks Mehrdad, yeah good question, this does look like a lot of files 🤔 I think all the actions are probably fine, since we do inject those patches there during the release process.  The `.dep-versions` and `doc/requirements.txt` are also updated according to the release guide. For the Makefile and setup.py I don't, we could leave those out for now and see later? The linux scripts are gone now (🎉), so that one can be removed :)